deployment/environment.go

```diff
@@ -50,12 +50,26 @@ type OffchainClient interface {
 type Chain struct {
 	// Selectors used as canonical chain identifier.
 	Selector uint64
+	name     string
 	Client   OnchainClient
 	// Note the Sign function can be abstract supporting a variety of key storage mechanisms (e.g. KMS etc).
 	DeployerKey *bind.TransactOpts
 	Confirm     func(tx *types.Transaction) (uint64, error)
 }
 
+func (c Chain) Name() string {
+	return c.name
```

Comment:
oh I was thinking of just inlining 
```
func (c Chain) Name() string {
	chainInfo, err := ChainInfo(c.Selector)
	if err != nil {
		// we should never get here, if the selector is invalid it should not be in the environment
		panic(err)
	}
	return chainInfo.Name
}
```
its just a map lookup inside there anyways